src/web_server.rs

```diff
@@ -116,42 +128,35 @@ pub async fn run_actix_server() -> std::io::Result<()> {
         // )
         // .service(web::resource("/manual").route(web::post().to(index_manual)))
         // .service(web::resource("/").route(web::post().to(index)))
-    })
-    .bind_rustls_0_23(format!("{}:{}", config_data.server.host, config_data.server.port), config)?
-    .run()
-    .await
+    });
+
+    let port = u16::from(config_data.server.port);
+    let server = match (tls, &config_data.server.host) {
```

Comment:
I am not very happy with this part. But implementing `std::net::ToSocketAddrs` trait for `Host` doesn't seem to be trivial as the `ToSocketAddrs` impl for `IpAddr` v/s `String` return different types https://doc.rust-lang.org/std/net/trait.ToSocketAddrs.html#implementors